### ISSUE TITLE
generate_dense_embeddings.py doesn't load custom TSV?

### ISSUE THREAD

**Issue by lukesalamone** [facebookresearch/ParlAI#3868]
**Bug description**
I am trying to generate embeddings for `bart_fid_rag` with a custom TSV but the `generate_dense_embeddings.py` script is loading wiki data instead.

**Reproduction steps**
1. Copy `ParlAI/data/models/hallucination/wiki_passages/psgs_w100.tsv` into `ParlAI/parlai/agents/rag/scripts/`
```
cd ParlAI/parlai/agents/rag/scripts/
cp ParlAI/data/models/hallucination/wiki_passages/psgs_w100.tsv passages.tsv
```

This simulates using a custom TSV file.

2. Run the following from `ParlAI/parlai/agents/rag/scripts/`

```
/usr/bin/python generate_dense_embeddings.py \
-mf zoo:hallucination/bart_fid_rag/model \
--dpr-model False \
--passages-file passages.tsv \
--outfile emb_bart_fid_rag/out \
--num-shards 1 \
--shard-id 0 \
-bs 32
```

**Expected behavior**
I don't expect the script to load `wiki_passages/psgs_w100.tsv` at all. I also don't expect it to load the index from `wiki_index_compressed/compressed_pq`. I'm not sure if my data is being read at all...

**Logs**
Please paste the command line output:

```
13:30:17 | Overriding opt["interactive_candidates"] to inline (previously: fixed)
13:30:17 | Using CUDA
13:30:17 | loading dictionary from /root/ParlAI/data/models/hallucination/bart_fid_rag/model.dict
13:30:17 | num words = 50264
13:30:17 | Fid: full interactive mode on.
13:30:19 | Loading index from /root/ParlAI/data/models/hallucination/wiki_index_compressed/compressed_pq
13:30:20 | Loaded index of type <faiss.swigfaiss.IndexIVFPQ; proxy of <Swig Object of type 'faiss::IndexIVFPQ *' at 0x7fb9abe86b70> > and size 21015300
13:30:23 | Reading data from: /root/ParlAI/data/models/hallucination/wiki_passages/psgs_w100.tsv
```

**Additional context**
Python 3.8.10

The script eventually fails with error

```
Traceback (most recent call last):
  File "generate_dense_embeddings.py", line 213, in <module>
    Generator.main()
  File "/root/ParlAI/parlai/core/script.py", line 129, in main
    return cls._run_args(None)
  File "/root/ParlAI/parlai/core/script.py", line 101, in _run_args
    return cls._run_from_parser_and_opt(opt, parser)
  File "/root/ParlAI/parlai/core/script.py", line 108, in _run_from_parser_and_opt
    return script.run()
  File "generate_dense_embeddings.py", line 102, in run
    assert hasattr(model, 'encoder_cand') or hasattr(model, 'cand_encoder')
AssertionError
```

**Comment by lukesalamone** [comment]
@klshuster

**Comment by klshuster** [comment]
Hi there, you've identified both a bug as well as some missing information in the dense embeddings instructions. The fix will require that you specify `--dpr-model True` to indicate you're generating the embeddings with a DPR-based retriever.

**Comment by lukesalamone** [comment]
Hey thanks for getting back to me. I tried `--dpr-model True` but I receive an exception pretty quickly:

```
Traceback (most recent call last):
  File "generate_dense_embeddings.py", line 213, in <module>
    Generator.main()
  File "/root/ParlAI/parlai/core/script.py", line 129, in main
    return cls._run_args(None)
  File "/root/ParlAI/parlai/core/script.py", line 101, in _run_args
    return cls._run_from_parser_and_opt(opt, parser)
  File "/root/ParlAI/parlai/core/script.py", line 108, in _run_from_parser_and_opt
    return script.run()
  File "generate_dense_embeddings.py", line 98, in run
    agent = create_agent(Opt(overrides))
  File "/root/ParlAI/parlai/core/agents.py", line 403, in create_agent
    model = create_agent_from_opt_file(opt)
  File "/root/ParlAI/parlai/core/agents.py", line 356, in create_agent_from_opt_file
    return model_class(opt_from_file)
  File "/root/ParlAI/parlai/agents/rag/dpr.py", line 55, in __init__
    super().__init__(opt, shared)
  File "/root/ParlAI/parlai/core/torch_ranker_agent.py", line 199, in __init__
    self.model = self.build_model()
  File "/root/ParlAI/parlai/agents/rag/dpr.py", line 58, in build_model
    return DPRModel(self.opt, self.dict)
  File "/root/ParlAI/parlai/agents/transformer/modules/mem_net.py", line 76, in __init__
    self.context_encoder = self.build_context_encoder(
  File "/root/ParlAI/parlai/agents/rag/dpr.py", line 229, in build_context_encoder
    return DprQueryEncoder(opt, dpr_model=query_model, pretrained_path=query_path)
  File "/root/ParlAI/parlai/agents/rag/dpr.py", line 170, in __init__
    super().__init__(*args, **kwargs)
  File "/root/ParlAI/parlai/agents/rag/dpr.py", line 134, in __init__
    self._load_state(dpr_model, pretrained_path, encoder_type)
  File "/root/ParlAI/parlai/agents/rag/dpr.py", line 148, in _load_state
    state_dict = BertConversionUtils.load_bert_state(
  File "/root/ParlAI/parlai/agents/rag/conversion_utils.py", line 57, in load_bert_state
    BertConversionUtils.load_dpr_model(
  File "/root/ParlAI/parlai/agents/rag/conversion_utils.py", line 93, in load_dpr_model
    for (key, value) in saved_state['model_dict'].items()
KeyError: 'model_dict'
```

Maybe my model file is corrupted?

**Comment by klshuster** [comment]
No, there is indeed a bug that needs to be fixed (see referenced PR #3869). In the meantime, feel free to specify `--model-file zoo:hallucination/bart_rag_token/model` as it will generate the same embeddings as the `bart_fid_rag` model

**Comment by lukesalamone** [comment]
Ah, I see. Thanks!
